design/authenticators/authn_gcp/phase_2/authn_gcp_solution_design.md

```diff
@@ -0,0 +1,252 @@
+    function_url = f'https://{REGION}-{PROJECT_ID}.cloudfunctions.net/{RECEIVING_FUNCTION}'
+    token_full_url = \
+    f'http://metadata/computeMetadata/v1/instance/service-accounts/default/identity?audience={AUDIENCE}'
+    token_headers = {'Metadata-Flavor': 'Google'}
+    
+    def fetch_token(request):
+    # Fetch the token
+    token_response = requests.get(token_full_url, headers=token_headers)
+    jwt = token_response.text
+    return jwt
+```
+
+### Access GCP authenticator
+Same as 1st phase
+
+### Backwards compatibility
+None because this is a new feature.
+
+### Performance
+Same as 1st phase:
+GCP authenticator performance should conform with our other authenticators with an average call time of 1 second.
+
+### Affected Components
+- Conjur
+
+- DAP
+
+## Cross-team dependencies
+* Infra team - the capability to fetch cloud function JWT token as part of our CI pipeline 
+
+## Security
+None, we are relying the same mechanism as 1st phase
+
+## Test Plan
+[Link to confluence](https://ca-il-confluence.il.cyber-ark.com/display/rndp/Conjur+GCP+authenticator+2nd+phase+-+Test+plan#/) 
+
+## Logs
+Will be changed as part of the implementation stage
+
+### Audit 
+Same logic like all other authenticators.
+
+## Documentation
+* Document examples of how to fetch google identity in google cloud functions
+* Document the solution supported use cases, logs and supportability section 
+
+
+## Automation Design
+Please read [solution design 1st phase Automation section](../phase_1/authn_gcp_solution_design.md) 
+
+In a similar approach we took in 1st phase, we will add the capability to create cloud functions tokens,
+in the jenkins machine as part of the CI process and will inject them to the cucumber flow to be used in the integrations tests.
+
+
+## Open questions
+
+## Delivery plan
+**Merging to master strategy**
+
+In this feature we are prefer to work with a feature branch instead of small PRs to master,
+Since we are planning to deliver it before the planned release version.
+
+### Total EE27
+
+### Parallelism
+The feature can be paralleled to 2 team members all the way
+
+#### Ramp up EE0
+already covered in research phase
+
+### Infrastructure and preparations EE4
+1. Create side branch **E0**
+    1.1 Create side branch from 11.7 release branch
+    1.2 Cherry pick commit of changing GCP name ([commit id](https://github.com/cyberark/conjur/commit/601db2781a08b5fb1b15bb262d846239b486092a))
+1. Support Google cloud functions in dev/start script **EE2**
+2. Update jenkins file in DAP and OSS to support Google cloud functions tokens **E1**
+3. Release from side branch **E1**
+    4.1 Update Conjur and DAP versions 
+    4.2 Release DAP as CA from side branch 
+
+#### Designs EE0
+None.
+Note: LLD designs need will be decided at implementation level 
+
+#### Spikes and researches EE1
+1. Does google app engine will be also supported? **E1**
```

Comment:
App engine is out of scope for this task

Reply:
EranH already did most of the work, we will not spend more then 2 hours on this assignment